includes/class-wc-stripe-webhook-handler.php

```
@@ -1827,6 +1844,128 @@ protected function handle_deferred_payment_intent_succeeded( $order, $intent_id
+		$reason           = __( 'Payment received in Stripe after the order was cancelled by the customer. Automatically refunded.', 'woocommerce-gateway-stripe' );
+		$result           = $this->process_refund( $order->get_id(), $is_authorization ? null : $order->get_total(), $reason );
+
+		if ( is_wp_error( $result ) ) {
+			$order->add_order_note(
+				sprintf(
+					/* translators: %s: Stripe error message. */
+					__( 'Stripe took a payment after this order was cancelled, but the automatic refund failed: %s. Please refund it manually in the Stripe dashboard.', 'woocommerce-gateway-stripe' ),
+					$result->get_error_message()
+				)
+			);
+			WC_Stripe_Logger::error( "Failed to refund cancelled order {$order->get_id()} (PaymentIntent {$intent_id}) after a late Stripe payment: {$result->get_error_message()}" );
```

PHPMD (2.15.0)

[error] Avoid using static access to class 'WC_Stripe_Logger' in method 'handle_deferred_payment_for_cancelled_order'. (undefined)

(StaticAccess)